CONTRIBUTING.md

```diff
@@ -1,7 +1,135 @@
 # Contributing Guidelines
 The OSM project accepts contributions via GitHub pull requests. This document outlines the requirements for contributing to this project.
 
-## Sign Your Work
+## Pull request workflow
+
+The following sections describe how to contribute code by opening a pull request.
+
+### 1 Fork the [Open Service Mesh](https://github.com/openservicemesh/osm) repository
+
+1. Visit `https://github.com/openservicemesh/osm`.
+1. Click the `Fork` button.
+
+### 2 Clone your fork
+Set `username` to match your Github username:
+```
+export username=<github username>
+```
+
+Clone and set up your fork:
+```sh
+git clone git@github.com:$username/osm.git
+
+cd osm
+git remote add upstream git@github.com:openservicemesh/osm.git
+
+# Block accidental pushes to upstream's main branch
+git remote set-url --push upstream no_push
+
+# Verify your remote
+git remote -v
+```
+
+### 3 Git branch
+
+Get your local `main` branch up to date with upstream's `main` branch:
+```sh
+git fetch upstream
+git checkout main
+git rebase upstream/main
+```
+
+Create a local branch for your development:
+```sh
+git checkout -b feature
+```
+
+Keep your branch up to date during development:
+```sh
+git fetch upstream
+git rebase upstream main
+
+# or: git pull --rebase upstream main
+```
+
+### 4 Commit
+Make code changes on the `feature` branch and commit them with your signature
+```sh
+git commit -s
+```
+
+Follow the [commit style guidelines](#commit-style-guideline).
+
+Make sure to squash your commits before opening a pull request. This is preferred so that your pull request can be merged as is without requesting to be squashed before merge if possible.
+
+### 5 Push
+Push your branch to your remote fork:
+```sh
+git push -f <remote name> <branch name>
+```
+
+### 6 Open a pull request
+1. Visit your fork at `https://github.com/$username/osm`.
+1. Open a pull request from your `feature` branch using the `Compare & Pull Request` button.
+1. Fill the pull request template and provide enough description so that it can be reviewed.
+
+If your pull request is not ready to be reviewed, open it as a draft and add the `wip` label.
```

Comment:
I don't think external contributors can add labels unless something has changed. May want to recommend they make the PR a "Draft PR" instead